src/codeflare_sdk/ray/rayjobs/test_rayjob.py

```diff
@@ -992,3 +993,729 @@ def test_build_ray_cluster_spec_with_gcs_ft(mocker):
+
+    # Should return early without processing
+    rayjob._process_script_and_imports(
+        "test.py", scripts, "/home/ray/scripts", processed_files
+    )
+
+    # Should remain unchanged
+    assert len(scripts) == 0
+    assert processed_files == {"test.py"}
+
+
+def test_find_local_imports_syntax_error(mocker):
+    """Test _find_local_imports handles syntax errors gracefully."""
+    mocker.patch("kubernetes.config.load_kube_config")
+    mock_api_class = mocker.patch("codeflare_sdk.ray.rayjobs.rayjob.RayjobApi")
+
+    rayjob = RayJob(
+        job_name="test-job",
+        cluster_name="existing-cluster",
+        entrypoint="python test.py",
+        namespace="test-namespace",
+    )
+
+    # Invalid Python syntax
+    invalid_script_content = "import helper\ndef invalid_syntax("
+
+    mock_callback = mocker.Mock()
+
+    # Should handle syntax error gracefully
+    rayjob._find_local_imports(invalid_script_content, "test.py", mock_callback)
+
+    # Callback should not be called due to syntax error
+    mock_callback.assert_not_called()
+
+
+def test_create_configmap_api_error_non_409(mocker):
+    """Test _create_configmap_from_spec handles non-409 API errors."""
+    mocker.patch("kubernetes.config.load_kube_config")
+    mock_api_class = mocker.patch("codeflare_sdk.ray.rayjobs.rayjob.RayjobApi")
+
+    # Mock Kubernetes API with 500 error
+    mock_k8s_api = mocker.patch("kubernetes.client.CoreV1Api")
+    mock_api_instance = mocker.Mock()
+    mock_k8s_api.return_value = mock_api_instance
+
+    from kubernetes.client import ApiException
+
+    mock_api_instance.create_namespaced_config_map.side_effect = ApiException(
+        status=500
+    )
+
+    mocker.patch("codeflare_sdk.ray.rayjobs.rayjob.get_api_client")
+
+    rayjob = RayJob(
+        job_name="test-job",
+        cluster_name="existing-cluster",
+        entrypoint="python test.py",
+        namespace="test-namespace",
+    )
+
+    configmap_spec = {
+        "apiVersion": "v1",
+        "kind": "ConfigMap",
+        "metadata": {"name": "test-scripts", "namespace": "test-namespace"},
+        "data": {"test.py": "print('test')"},
+    }
+
+    # Should raise RuntimeError for non-409 API errors
+    with pytest.raises(RuntimeError, match="Failed to create ConfigMap"):
+        rayjob._create_configmap_from_spec(configmap_spec)
+
+
+def test_update_existing_cluster_get_cluster_error(mocker):
+    """Test _update_existing_cluster_for_scripts handles get cluster errors."""
+    mocker.patch("kubernetes.config.load_kube_config")
+    mock_api_class = mocker.patch("codeflare_sdk.ray.rayjobs.rayjob.RayjobApi")
+
+    # Mock CustomObjectsApi with error
+    mock_custom_api = mocker.patch("kubernetes.client.CustomObjectsApi")
+    mock_api_instance = mocker.Mock()
+    mock_custom_api.return_value = mock_api_instance
+
+    from kubernetes.client import ApiException
+
+    mock_api_instance.get_namespaced_custom_object.side_effect = ApiException(
```

Comment:
This will need to be changed based on the api change above. Can just mock the cluster api instead